recirq/quantum_chess/quantum_board.py

```diff
@@ -257,40 +244,53 @@ def sample(self, num_samples: int) -> List[int]:
             rtn = rtn + samples
         return rtn[:num_samples]
 
+    def generate_accumulations(self, repetitions: int = 1000):
+        """ Samples the state and generates the accumulated 
+        probabilities, empty_squares, and full_squares
+        """
+        self.probabilities = [0] * 64
+        self.full_squares = (1 << 64) - 1
+        self.empty_squares = (1 << 64) - 1
+
+        samples = self.sample(repetitions)
+        for sample in samples:
+            for bit in range(64):
+                if nth_bit_of(bit, sample):
+                    self.probabilities[bit] += 1
+                    self.empty_squares = set_nth_bit(bit,self.empty_squares,0)
+                else:
+                    self.full_squares = set_nth_bit(bit,self.full_squares,0)
+
+        for bit in range(64):
+            self.probabilities[bit] = float(self.probabilities[bit]) / float(repetitions)
+        
+        self.accumulations_valid = True
```

Comment:
We should also add tests for this or make it a private function (add underscore)

Reply:
Made it private